src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -109,4 +110,71 @@ def get_date(file):
         except Exception as e:
             print(f"An error occurred: {e}")
 
-        outputs.set_value("corpus_table", output_table)
\ No newline at end of file
+        outputs.set_value("corpus_table", output_table)
+
+class CorpusDistTime(KiaraModule):
+    """
+    This module aggregates a table by day, month or year from a corpus table that contains a date column. It returns the distribution over time, which can be used for display purposes, such as visualization.
+
+    Dependencies:
+    - polars: https://www.pola.rs/
+    - pyarrow: https://arrow.apache.org/docs/python/
+    - duckdb: https://duckdb.org/
+    
+    """
+
+    _module_type_name = "topic_modelling.time_dist"
+
+    def create_inputs_schema(self):
+        
+        return {
+             "periodicity": {
```

Comment:
You never test this value for correctness in the `process` method, what if a user specifies 'century', for example? You can either add a test there, or limit the options a user can specify by:

```python
  "type": "string",
  "type_config": {
     "allowed_strings": ["day", "month", "year"]
  }
```

In such cases, you can always look up the data type source code to see if there are any config options, like for example: https://github.com/DHARPA-Project/kiara/blob/0402ff03d3c479e378b78bdc278f8b9217516922/src/kiara/data_types/included_core_types/__init__.py#L239

This shows that the data type has a config class attached to it, and the config class has one option, 'allowed_strings'.